x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/utils/indices.ts

```diff
@@ -27,9 +27,15 @@ import {
 } from '../types';
 
 export function isConnectorIndex(
-  index: ElasticsearchIndexWithIngestion | undefined
+  index: ElasticsearchIndexWithIngestion | undefined,
+  includeCrawler: boolean = false
 ): index is ConnectorIndex {
   const connectorIndex = index as ConnectorIndex;
+
+  if (includeCrawler) {
+    return !!connectorIndex?.connector;
+  }
+
```

Comment:
These changes are necessary only because we haven't got a migration yet. Once we have a migration we can refactor most checks as every crawler will be a connector.